File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(auto& itr:nums){
            mp[itr]++;
        }
        int majority=nums.size()/2;
        for(auto& it:mp){
            if(it.second>majority){
                return it.first;
            }
        }
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.py
from typing import List

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counts = {}                       # don’t shadow built-in name `map`
        threshold = len(nums) // 2       # majority threshold
        
        # 1) Build counts
        for num in nums:
            counts[num] = counts.get(num, 0) + 1
        
        # 2) Find the element whose count exceeds n/2
        for num, cnt in counts.items():  # iterate over key/value pairs
            if cnt > threshold:
                return num               # return immediately
        
        # problem guarantees a majority element, but to satisfy function signature:
        return -1
